e2e-tests/playwright/e2e/authProviders/github.spec.ts

```diff
@@ -0,0 +1,276 @@
+      "AUTH_PROVIDERS_GH_ORG_CLIENT_SECRET",
+      process.env.AUTH_PROVIDERS_GH_ORG_CLIENT_SECRET,
+    );
+    deployment.addSecretData(
+      "AUTH_PROVIDERS_GH_ORG_CLIENT_ID",
+      process.env.AUTH_PROVIDERS_GH_ORG_CLIENT_ID,
+    );
+    deployment.addSecretData(
+      "AUTH_PROVIDERS_GH_ORG_APP_ID",
+      process.env.AUTH_PROVIDERS_GH_ORG_APP_ID,
+    );
+    deployment.addSecretData(
+      "AUTH_PROVIDERS_GH_ORG1_PRIVATE_KEY",
+      process.env.AUTH_PROVIDERS_GH_ORG1_PRIVATE_KEY,
+    );
+    deployment.addSecretData(
+      "AUTH_PROVIDERS_GH_ORG_WEBHOOK_SECRET",
+      process.env.AUTH_PROVIDERS_GH_ORG_WEBHOOK_SECRET,
+    );
+
+    await deployment.createSecret();
+
+    // enable keycloak login with ingestion
+    await deployment.enableGithubLoginWithIngestion();
+    await deployment.updateAllConfigs();
+
+    // create backstage deployment and wait for it to be ready
+    await deployment.createBackstageDeployment();
+    await deployment.waitForDeploymentReady();
+
+    // wait for rhdh first sync and portal to be reachable
+    await deployment.waitForSynced();
+  });
+
+  test.beforeEach(async () => {
+    test.info().setTimeout(600 * 1000);
+    console.log(
+      `Running test case ${test.info().title} - Attempt #${test.info().retry}`,
+    );
+  });
+
+  test("Login with Github default resolver", async () => {
+    const login = await common.githubLogin(
+      "rhdhqeauthadmin",
+      process.env.AUTH_PROVIDERS_GH_USER_PASSWORD,
+      process.env.AUTH_PROVIDERS_GH_ADMIN_2FA,
+    );
+    expect(login).toBe("Login successful");
+
+    await page.goto("/settings");
+    await uiHelper.verifyHeading("RHDH QE Admin");
+    await common.signOut();
+    await context.clearCookies();
+  });
+
+  test("Login with Github emailMatchingUserEntityProfileEmail resolver", async () => {
+    //A common sign-in resolver that looks up the user using the local part of their email address as the entity name.
+    await deployment.setGithubResolver(
+      "emailMatchingUserEntityProfileEmail",
+      false,
+    );
+    await deployment.updateAllConfigs();
+    await deployment.restartLocalDeployment();
+    await page.waitForTimeout(3000);
+    await deployment.waitForDeploymentReady();
+
+    // wait for rhdh first sync and portal to be reachable
+    await deployment.waitForSynced();
+
+    const login = await common.githubLogin(
+      "rhdhqeauth1",
+      process.env.AUTH_PROVIDERS_GH_USER_PASSWORD,
+      process.env.AUTH_PROVIDERS_GH_USER_2FA,
+    );
+    expect(login).toBe("Login successful");
+
+    await uiHelper.verifyAlertErrorMessage(
+      /Login failed; caused by Error: Failed to sign-in, unable to resolve user identity. Please verify that your catalog contains the expected User entities that would match your configured sign-in resolver. For non-production environments, manually provision the user or disable the user provisioning requirement by setting the `dangerouslyAllowSignInWithoutUserInCatalog` option./,
```

Comment:
maybe we can store this in a global constant so it's accessible by all auth provider tests?  If the message ever changes, we can change it in one place

Reply:
Yep, that's better. I moved to the constants file. Thanks!